packages/@react-aria/calendar/src/useCalendarCell.ts

```diff
@@ -283,17 +283,27 @@ export function useCalendarCell(props: AriaCalendarCellProps, state: CalendarSta
 
   // Focus the button in the DOM when the state updates.
   useEffect(() => {
+    let raf;
     if (isFocused && ref.current) {
       focusWithoutScrolling(ref.current);
 
       // Scroll into view if navigating with a keyboard, otherwise
       // try not to shift the view under the user's mouse/finger.
-      // Only scroll the direct scroll parent, not the whole page, so
-      // we don't scroll to the bottom when opening date picker popover.
+      // If in a overlay, scrollIntoViewport will only cause scrolling
+      // up to the overlay scroll body to prevent overlay shifting
       if (getInteractionModality() !== 'pointer') {
-        scrollIntoView(getScrollParent(ref.current) as HTMLElement, ref.current);
+        // If this calendar is in a popover, we need to wait till it is fully positioned before attempting to
+        // scroll it into view, otherwise it won't be brought into view properly
+        raf = requestAnimationFrame(() => {
```

Comment:
I don't believe this runs into the same problem that the FocusScope bug was running into right? This uses a variable initialized in the useEffect instead of setting it to a ref outside of the useEffect. The new raf is only created if any of the dependencies change and the cleanup will cancel the pre-existing raf before the new one is then created.

Reply:
o, yep, you are right. I didn't notice it was inside the useEffect